packages/boot/test/bootstrapTests/price-feed-replace.test.ts

```diff
@@ -0,0 +1,200 @@
+/**
+ * @file  The goal of this test is  to see that the
+ * upgrade scripts re-wire all the contracts so new auctions and
+ * price feeds are connected to vaults correctly.
+ *
+ * 1. enter a bid
+ * 2. force prices to drop so a vault liquidates
+ * 3. verify that the bidder gets the liquidated assets.
+ */
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import type { TestFn } from 'ava';
+import { ScheduleNotification } from '@agoric/inter-protocol/src/auction/scheduler.js';
+import { NonNullish } from '@agoric/internal';
+import { eventLoopIteration } from '@agoric/internal/src/testing-utils.js';
+import { Fail } from '@endo/errors';
+
+import {
+  LiquidationTestContext,
+  likePayouts,
+  makeLiquidationTestContext,
+  scale6,
+  LiquidationSetup,
+} from '../../tools/liquidation.js';
+import {
+  updateVaultDirectorParams,
+  updateVaultManagerParams,
+} from '../tools/changeVaultParams';
+
+const test = anyTest as TestFn<LiquidationTestContext>;
+test.before(
+  async t =>
+    (t.context = await makeLiquidationTestContext(t, { env: process.env })),
+);
+test.after.always(t => t.context.shutdown());
+
+const collateralBrandKey = 'ATOM';
+const managerIndex = 0;
+
+const setup: LiquidationSetup = {
+  vaults: [{ atom: 15, ist: 100, debt: 100.5 }],
+  bids: [{ give: '20IST', discount: 0.1 }],
+  price: {
+    starting: 12.34,
+    trigger: 9.99,
+  },
+  auction: {
+    start: { collateral: 15, debt: 100.5 },
+    end: { collateral: 9.659301, debt: 0 },
+  },
+};
+
+const outcome = {
+  bids: [{ payouts: { Bid: 0, Collateral: 2.224446 } }],
+};
+
+test.serial('setupVaults; run replace-price-feeds proposals', async t => {
+  const {
+    agoricNamesRemotes,
+    buildProposal,
+    evalProposal,
+    priceFeedDrivers,
+    refreshAgoricNamesRemotes,
+    setupVaults,
+    governanceDriver: gd,
+  } = t.context;
+
+  await setupVaults(collateralBrandKey, managerIndex, setup);
+
+  const instancePre = agoricNamesRemotes.instance['ATOM-USD price feed'];
+
+  const priceFeedBuilder =
+    '@agoric/builders/scripts/inter-protocol/updatePriceFeeds.js';
+  t.log('building', priceFeedBuilder);
+  const brandName = collateralBrandKey;
+
+  const { ATOM } = agoricNamesRemotes.brand;
+  ATOM || Fail`ATOM missing from agoricNames`;
+  await updateVaultManagerParams(t, gd, ATOM, 50_000_000n);
+
+  const SOME_GUI = 'someGUIHASH';
+  await updateVaultDirectorParams(t, gd, SOME_GUI);
+
+  t.log('building all relevant CoreEvals');
+  const coreEvals = await Promise.all([
+    buildProposal(priceFeedBuilder, ['UNRELEASED_main']),
+    buildProposal('@agoric/builders/scripts/vats/upgradeVaults.js'),
+    buildProposal('@agoric/builders/scripts/vats/add-auction.js'),
+  ]);
```

Comment:
The order in this promise all could make a difference somewhere. It would be good to do the same in production; if we're using cosgov, that means a human is choosing the order.